### PR TITLE
RHMAP-8410 install oc tools

### PR DESCRIPTION
**Pull request by jessesarn**
This adds the OSE client tools which provide oc which will be especially useful for its exec functionality when creating nagios checks with probe into the running pods.

### CODE REVIEW

**Comment by jessesarn** [comment]
@rhcarvalho, @mikenairn Could one of you please review when you get a chance?  My other PR's will be based off this branch since they rely on oc/jq to be installed.

File: Dockerfile [hunk 1]
```diff
@@ -2,10 +2,13 @@ FROM centos:centos7
 
 EXPOSE 8080
 
+RUN curl --retry 999 --retry-max-time 0 -sSL https://github.com/openshift/origin/releases/download/v1.2.0/openshift-origin-client-tools-v1.2.0-2e62fab-linux-64bit.tar.gz | tar xzv && \
+  mv openshift-origin-*/* /usr/bin/
```
Comment (mikenairn) [inline Dockerfile]: @rhcarvalho Does this seem like a reasonable way to install the client?

Comment (mikenairn) [inline Dockerfile]: @rhcarvalho What's the best practise with multiple RUN instructions in a Dockerfile? Is it best to keep them separate producing more layers, or bundle them all up in one RUN instruction? i.e Should this be added to the one below?

File: Dockerfile [hunk 2]
```diff
@@ -2,10 +2,13 @@ FROM centos:centos7
...
+
 ADD sendEmail-epel-7.repo /etc/yum.repos.d/
 
 RUN yum install -y epel-release && \
-    INSTALL_PKGS="httpd nagios telnet supervisor python-jinja2 nagios-plugins-all sendEmail perl-Net-SSLeay perl-IO-Socket-SSL" && \
+    INSTALL_PKGS="httpd nagios telnet supervisor python-jinja2 nagios-plugins-all sendEmail perl-Net-SSLeay perl-IO-Socket-SSL jq" && \
```
Comment (mikenairn) [inline Dockerfile]: We should keep @vibe13 in the loop when adding new packages here.

File: Dockerfile (hunk 1 above)
Comment (rhcarvalho) [inline Dockerfile]: > Does this seem like a reasonable way to install the client?

I think its best to install via yum, Andrea will thank us. The package is called openshift-cli, IIRC.

Comment (rhcarvalho) [inline Dockerfile]: > What's the best practise with multiple RUN instructions in a Dockerfile?

The best practice is to combine them as much as possible.
In general, we should look for having the least number of layers possible, so this practice extends to the use of other instructions as well, such as EXPOSE, ENV, ADD, ...

**Comment by rhcarvalho** [comment]
@jessesarn please rebase this and it should be just about adding `oc`, right?
`jq` should be already there now.

**Comment by jessesarn** [comment]
Reduced to just the oc install via curl.

**Comment by mikenairn** [comment]
Looks good to me.

File: Dockerfile [hunk 3]
```diff
@@ -9,6 +9,8 @@ RUN yum install -y epel-release && \
     yum -y --setopt=tsflags=nodocs install $INSTALL_PKGS && \
     rpm -V $INSTALL_PKGS && \
     yum clean all && \
+    curl --retry 999 --retry-max-time 0 -sSL https://github.com/openshift/origin/releases/download/v1.2.0/openshift-origin-client-tools-v1.2.0-2e62fab-linux-64bit.tar.gz | tar xzv && \
+    mv openshift-origin-*/* /usr/bin/ \
```
Comment (rhcarvalho) [inline Dockerfile]: Isn't it missing a `&&` here?

**Pull request merged**